plugins/module_utils/network/sonic/config/bgp_communities/bgp_communities.py

```diff
@@ -361,8 +419,165 @@ def get_modify_bgp_community_requests(self, commands, have):
+                else:
+                    if have_conf['permit']:
+                        delete_conf['permit'] = True
+
+                if 'match' not in conf:
+                    if have_conf['match'] and have_conf['match'] == "ALL":
+                        delete_conf['match'] = have_conf['match']
+                else:
+                    if have_conf['match'] != conf['match']:
+                        delete_conf['match'] = have_conf['match']
+                    add_conf['match'] = conf['match']
+
+                if 'aann' not in conf:
+                    if have_conf['aann']:
+                        delete_conf['aann'] = have_conf['aann']
+                else:
+                    if conf['aann']:
+                        aann = list(map(str, conf['aann']))
+                        aann.sort()
+                        if have_conf['aann']:
+                            have_conf['aann'].sort()
+                            if have_conf['aann'] != aann:
+                                delete_conf['aann'] = have_conf['aann']
+                        add_conf['aann'] = aann
+                    else:
+                        if have_conf['aann']:
+                            delete_conf['aann'] = have_conf['aann']
+
+                if have_conf['members'] and 'regex' in have_conf['members']:
+                    if have_conf['members']['regex']:
+                        have_conf['members']['regex'].sort()
+                if 'members' not in conf:
+                    if have_conf['members'] and 'regex' in have_conf['members']:
+                        delete_conf['members'] = have_conf['members']
+                else:
+                    if conf['members']:
+                        if 'regex' in conf['members']:
+                            if conf['members']['regex']:
+                                members = list(map(str, conf['members']['regex']))
+                                members.sort()
+                                if have_conf['members'] and 'regex' in have_conf['members']:
+                                    if have_conf['members']['regex'] != members:
+                                        delete_conf['members'] = have_conf['members']
+                                add_conf['members'] = {'regex': members}
+                            else:
+                                if have_conf['members'] and 'regex' in have_conf['members']:
+                                    delete_conf['members'] = have_conf['members']
+                        else:
+                            if have_conf['members'] and 'regex' in have_conf['members']:
+                                delete_conf['members'] = have_conf['members']
+                    else:
+                        if have_conf['members'] and 'regex' in have_conf['members']:
+                            delete_conf['members'] = have_conf['members']
+
+                add_conf['name'] = name
+                commands_add.append(add_conf)
```

Comment:
Please update commands_add, only when there is config to add for that community list.
Doing so would help in removing an additional get_diff call in '_state_replaced' and '_state_overridden'.